pkg/agent/plugin/workloadattestor/docker/docker_posix.go

```diff
@@ -76,6 +76,14 @@ func getContainerIDFromCGroups(finder cgroup.ContainerIDFinder, cgroups []cgroup
 		switch {
 		case containerID == "":
 			// This is the first container ID found so far.
+			// Try match on extracted path element to support cgroups v2 with
+			// systemd driver /proc values.
+			if v2match := dockerCGroupRE.FindStringSubmatch(candidate); v2match != nil {
+				containerID = v2match[1]
+				continue
+			}
+			// Assume candidate path element is already a container id from
+			// cgroupsfs driver /proc values.
```

Comment:
@azdagron 👍  or 👎 Is this a sane approach? If yes, I'll update docs around the `<id>` matcher to describe this behavior to make this shippable.

Reply:
> Unfortunately, I don't think this patch is ideal.

I agree it is potentially suboptimal, but it does work.

> The intention of the loop is to attempt to extract the container ID from each of the cgroups. Once a container ID has been encountered, we continue processing each cgroup to make sure that the container IDs extracted from each are consistent (if they aren't then our finder was too "lenient" and there is ambiguity on what the container ID actually is).

Under the systemd driver for `/proc`, there is only ever 1 cgroup at the top-level for each system unit so the loop will break at the first hit anyway under cgroups v2 with systemd driver. Effectively it already short-circuits on first hit and by applying the default matcher, we can still extract the ID from the systemd driver `/proc`.

> How is this patch any different in behavior than the current code when container_id_cgroup_matchers is not defined (and therefore the "finder" does an identical regex match to the one you added)?

1) It allows us to leave SPIRE Agent container_id_cgroup_matchers configuration unchanged so that we do not need to coordinate with a rollout of cgroups v2 with systemd driver.
2) It allows us to keep narrow matching to `/system.slice/*/<id>` in config after transition to cgroups v2 is complete.

> Is an agent deployed in an environment where it needs to support both custom cgroup matches and the default?

No single agent will need to support both at one time, but we don't want to need to care whether any given host is using cgroups v1 with cgroupsfs driver or cgroups v2 with sysemd driver.